From 6921c206918a2bd9882bbf77211d37a063233d95 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Thu, 8 Aug 2019 16:13:40 +0800
Subject: [PATCH] MLK-22099 usb: host: xhci: do warm reset for link state
 rxdect

As some USB3 port will lose link state while system sleep, then
the link state will be at rxdect after resume, we need a warm
reset to bring it back, so add the rxdect condition for CAS
missing.

Signed-off-by: Li Jun <jun.li@nxp.com>
---
 drivers/usb/host/xhci-hub.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/usb/host/xhci-hub.c
+++ b/drivers/usb/host/xhci-hub.c
@@ -1739,7 +1739,8 @@ static bool xhci_port_missing_cas_quirk(
 		return false;
 
 	if (((portsc & PORT_PLS_MASK) != XDEV_POLLING) &&
-	    ((portsc & PORT_PLS_MASK) != XDEV_COMP_MODE))
+	    ((portsc & PORT_PLS_MASK) != XDEV_COMP_MODE) &&
+	    ((portsc & PORT_PLS_MASK) != XDEV_RXDETECT))
 		return false;
 
 	/* clear wakeup/change bits, and do a warm port reset */
